docs/explanation/remote-build.rst

```diff
@@ -89,12 +89,20 @@ Current
    ``--platform`` and ``--build-for`` behave differently than they do for
    :ref:`lifecycle commands<reference-lifecycle-commands>`.
 
-``--platform`` or ``--build-for`` can only be provided when the ``platforms``
-or ``architectures`` keywords are not defined in the project metadata
-(`[12]`_).
+If the project file contains a ``platforms`` or ``architectures`` key, then
+``--platform`` or ``--build-for`` filter the build plan defined by that key.
+For more information about build plans and filtering, see
+:ref:`Build plans <build-plans>`.
 
-These keywords are mutually exclusive and must be a comma-separated list of
-debian architectures.
+If the project file doesn't contain a ``platforms`` or ``architectures`` entry,
+then ``--platform`` or ``--build-for`` define the architectures to build for.
+
+It the project file doesn't contain a ``platforms`` or ``architectures`` entry
+and ``--platform`` or ``--build-for`` are not provided, Snapcraft will build
+for the host system's architecture.
+
+These ``--platform`` and ``--build-for`` arguments are mutually exclusive and
```

Comment:
Why do we only accept --platform as debian architectures? Is it because of launchpad?

Reply:
Resolved via removing `--platform`.